library/init/meta/widget/basic.lean

```diff
@@ -117,19 +171,41 @@ inductive mouse_event_kind
 | on_mouse_enter
```

Comment:
I do not like that there are two ways to observe the mouse-over state now, where one of them is so fragile that it shouldn't be used (or is this fixed now?).  If the `on_mouse_enter` and `on_mouse_leave` events are still not robust, then maybe we should remove them.

Reply:
no they both have roles, since with_mouse can't emit actions. I think there are use cases for both.

Reply:
I'll have a go at refactoring interactive_expr to not use with_mouse. interactive_expr has to use on_mouse_enter and on_mouse_leave because hovering over a substring causes a change outside that dom element.